web/documentserver-example/csharp/Default.aspx.cs

```diff
@@ -284,9 +244,9 @@ public static string DocumentType(string fileName)
         {
             var ext = Path.GetExtension(fileName).ToLower();
 
-            if (FileType.ExtsDocument.Contains(ext)) return "word";  // word for text document extensions
-            if (FileType.ExtsSpreadsheet.Contains(ext)) return "cell";  // cell for spreadsheet extensions
-            if (FileType.ExtsPresentation.Contains(ext)) return "slide";  // slide for presentation extensions
+            if (FormatManager.DocumentExtensions().Contains(ext)) return "word";  // word for text document extensions
+            if (FormatManager.SpreadsheetExtensions().Contains(ext)) return "cell";  // cell for spreadsheet extensions
+            if (FormatManager.PresentationExtensions().Contains(ext)) return "slide";  // slide for presentation extensions
```

Comment:
Why do you call the All() method 3 times? Therefore the file "onlyoffice-docs-formats.json" is read 3 times.

Reply:
> Why do you call the All() method 3 times? Therefore the file "onlyoffice-docs-formats.json" is read 3 times.

I added format caching